actors/miner/tests/util.rs

```diff
@@ -457,17 +477,173 @@ impl ActorHarness {
         }
     }
 
+    pub fn make_pre_commit_params_v2(
+        &self,
+        sector_no: u64,
+        challenge: ChainEpoch,
+        expiration: ChainEpoch,
+        sector_deal_ids: Vec<DealID>,
+        unsealed_cid: Option<Cid>,
+    ) -> SectorPreCommitInfo {
+        SectorPreCommitInfo {
+            seal_proof: self.seal_proof_type,
+            sector_number: sector_no,
+            sealed_cid: make_sealed_cid(b"commr"),
+            seal_rand_epoch: challenge,
+            deal_ids: sector_deal_ids,
+            expiration,
+            unsealed_cid: CompactCommD::new(unsealed_cid),
+        }
+    }
+
     pub fn make_prove_commit_params(&self, sector_no: u64) -> ProveCommitSectorParams {
         ProveCommitSectorParams { sector_number: sector_no, proof: vec![0u8; 192] }
     }
 
+    pub fn pre_commit_sector_batch_v2(
+        &self,
+        rt: &mut MockRuntime,
+        params: PreCommitSectorBatchParams2,
+        first_for_miner: bool,
+        base_fee: &TokenAmount,
+    ) -> Result<RawBytes, ActorError> {
+        if !self.options.use_v2_pre_commit_and_replica_update {
```

Comment:
This is really weird. I think the only path in here is from `pre_commit_sector_batch`, when this option is true. The mutual "recursion" is hard to follow.

This has also led to a lot of duplication between the methods. Could you refactor down to just one method that checks this flag, but re-uses the logic that hasn't changed?

Reply:
I could but as the types of parameters that get passed are different, it is a bit annoying. I think there might be a nice way to do it using a generic.

Reply:
> The mutual "recursion" is hard to follow.

It isn't really mutual recursion, the methods forward to each other if "a wrong method given the harness settings" was called